report-generation-nlp.py

```diff
@@ -0,0 +1,329 @@
+import dgl
```

Comment:
Actually I think you should open another PR for this, so that we solve a smaller problem one at a time